rust/noosphere-gateway/src/handlers/v0alpha1/push.rs

```diff
@@ -50,23 +46,25 @@ where
     debug!("Invoking push route...");
 
     let sphere_identity = &request_body.sphere;
-
-    if sphere_identity != &gateway_scope.counterpart {
+    let counterpart = &gateway_scope.counterpart;
+    if sphere_identity != counterpart {
         return Err(StatusCode::FORBIDDEN);
     }
 
-    authority.try_authorize(&generate_capability(
-        &gateway_scope.counterpart,
-        SphereAbility::Push,
-    ))?;
+    authority
+        .try_authorize(
+            &mut gateway_sphere,
+            counterpart,
+            &generate_capability(counterpart.as_str(), SphereAbility::Push),
+        )
+        .await?;
 
     let gateway_push_routine = GatewayPushRoutine {
-        sphere_context,
+        gateway_sphere,
```

Comment:
This naming tripped me up in my last change (and I'm the original author for crying out loud), so thanks for clarifying it!

Reply:
hah, the spheres and contexts trip me up too